src/control/common/proto/error_test.go

```diff
@@ -148,27 +123,29 @@ func TestProto_AnnotateError(t *testing.T) {
 		t.Run(name, func(t *testing.T) {
 			aErr := proto.AnnotateError(tc.err)
 
-			var gotErr error
+			gotErr := proto.UnwrapError(status.Convert(aErr))
+			common.CmpErr(t, tc.expErr, gotErr)
+			if tc.expErr == nil {
+				return
+			}
+
 			switch tc.err.(type) {
```

Comment:
Is this switch block still needed? Looks like the expErr/gotErr comparison above should cover this, should it not?

Reply:
Hmm, good question. I think I cargo-culted my own code without really thinking about it. There was some goofy business in there before with the default case... Can probably clean this up to make it easier to maintain.